### PR TITLE
cve_scan: Use `envoy.dependency.cve_scan`

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/dependency/BUILD [hunk 1]
```diff
@@ -22,25 +22,16 @@ py_library(
...
-)
-
-py_binary(
-    name = "cve_scan_test",
-    srcs = ["cve_scan_test.py"],
-    data = [
-        ":cve_scan",
+    deps = [
+        ":utils",
+        "@envoy_repo",
```
Comment (phlax) [inline tools/dependency/BUILD:33]: not used

File: tools/dependency/cve_scan.py [hunk 2]
```diff
@@ -1,275 +1,97 @@
...
-import gzip
-import json
-import re
+# usage
+#
+# with bazel:
+#
+#  $ bazel run //tools/dependency:pip_check -- -h
+#
+#  $ bazel run //tools/dependency:pip_check
```
Comment (phlax) [inline tools/dependency/cve_scan.py:9]: fix this

**Comment by phlax** [comment]
demo CI run (with `ignored_cves` disabled):

https://dev.azure.com/cncf/envoy/_build/results?buildId=94816&view=logs&j=6f2cd4c8-2a7f-58b3-6ca1-df9e9e926610&t=fecae80b-e9b1-5bfa-068e-63d94be322a7&l=1892

**Comment by phlax** [comment]
a successful run (which normally openly happens as a chron task) looks like this:

https://dev.azure.com/cncf/envoy/_build/results?buildId=94830&view=logs&j=6f2cd4c8-2a7f-58b3-6ca1-df9e9e926610&t=fecae80b-e9b1-5bfa-068e-63d94be322a7&l=1588

File: tools/dependency/cve_scan.py [hunk 3]
```diff
@@ -1,275 +1,96 @@
...
-# (and near relatives). We also try a fuzzy match on version information.
+# usage
+#
+# with bazel:
+#
+#  $ bazel run //tools/dependency:cve_scan -- -h
+#
+#  $ bazel run //tools/dependency:cve_scan
+#
+# with pip:
```
Comment (phlax) [inline tools/dependency/cve_scan.py:11]: remove

File: tools/dependency/cve.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ ndist_url: https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-{year}.json.gz
 
 # These CVEs are false positives for the match heuristics. An explanation is
 # required when adding a new entry to this list as a comment.
-ignore:
+ignored_cves:
```
Comment (phlax) [inline tools/dependency/cve.yaml:9]: remove unrequired cve ignores

**Review by htuch — COMMENTED** [review]
Great. Maybe test and verify if removing one of the ignore CVEs causes CI failure.

**Comment by htuch** [comment]
Oh, you already did! Then happy to stamp once you have this ready fro review.

**Comment by phlax** [comment]
this should be ready for review - there is an unrelated CI fail which is hitting all branches atm 8/

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**